### DISCUSSION TITLE
Feature request: make it harder for an org member to create public repos

### DISCUSSION THREAD

**Discussion by ronny** [community/community#23038] [New to GitHub]
<p>This would reduce accidental creations of repos that are meant to be private in the first place. Often these repos contain confidential information, and sometimes even private keys (yes, it’s bad practice that should be discouraged, but we can significantly lower the rate of accidents!). </p>
<p>One potential way to achieve this is perhaps an additional preference at the org level. The repo creation UI can then present additional hoops/warnings/verifications to be able to create a public repo, or even deny it entirely (maybe you could make this access more granular in later iterations). Creation of private repo = business as usual.</p>
<p>Often, these accidents are not noticed immediately because the UI indicator for private and public repos is barely noticeable. Maybe this could also be changed based on the additional preference. If the org prefers private repos, then the public ones can be rendered with a significantly noticable different theme, or something along that line.</p>

**Comment by ronny** [comment]
<p>Like this <a href="https://help.github.com/en/enterprise/2.17/admin/user-management/restricting-repository-creation-in-your-instance" rel="nofollow noopener">https://help.github.com/en/enterprise/2.17/admin/user-management/restricting-repository-creation-in-your-instance</a> but for non-Enterprise orgs.</p>

**Comment by ronny** [comment]
<p>Actually, why is it Enterprise Cloud only feature?</p>
<p><a href="https://help.github.com/en/articles/restricting-repository-creation-in-your-organization" rel="nofollow noopener">https://help.github.com/en/articles/restricting-repository-creation-in-your-organization</a></p>

**Comment by BelHFL** [comment]
<p>Hi <a class="mention" href="/ronny">@ronny</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Cheers!</p>

**Comment by i-marsh** [comment, marked as answer]
<p>Hey <a class="mention" href="/ronny">@ronny</a> </p>
<p>I’m not sure if you are familiar with <a href="https://probot.github.io/" title="Probot - GitHub automation bot" rel="nofollow noopener">Probot</a>, but there is an existing automation that may help you out <a href="https://probot.github.io/apps/prevent-public-repos/" title="probot - prevent public repos" rel="nofollow noopener">here</a></p>
<p>Working with Probot is super easy and for me at least, a little addictive :) </p>
<p>To save you a click-through here’s a quick summary from the apps home page.</p>
<h2>Features</h2>
<ul>
<li>Can convert newly created Public Repos to Private</li>
<li>Can also be enabled for repos that switch visibility from private to public
<ul>
<li>Not enabled by default it is possible to restrict visibility changes to org owners <a href="https://help.github.com/articles/repository-permission-levels-for-an-organization/#changing-the-visibility-of-repositories" rel="nofollow noopener">directly in GitHub</a>.</li>
</ul>
</li>
<li>Will create an issue in the repo explaining the action</li>
<li>Monitor only mode will not change the repo visibility but will still create an issue</li>
<li>Can set configuration parameters by using YAML file set in a specific repo for the entire org</li>
<li>Can exclude certain repos</li>
<li>Can set a list of users/groups to cc on every issue created</li>
</ul>

**Comment by ronny** [comment]
<p>Hi i-marsh, that’s great! It’s not ideal, but probably the next best option. We just installed this automation today, and I think the team likes it so far. Thanks for the suggestion!</p>
